src/lib/rose_tree/rose_tree.mli

```diff
@@ -0,0 +1,7 @@
+type 'a t = T of 'a * 'a t list
```

Comment:
doccomments here too would be nice. I have some docs on the old ktree code around this